migrate_tool_main.pl

```diff
@@ -36,9 +47,108 @@
 my $RecoveryMode = "";
 my $DEBUG_MODE = 0;
 
+use Getopt::Long qw(GetOptions);
+
+my $mode;
+my $configfile;
+my $siteindex;
+my $pubprofile;
+my @params;
+
+
 eval
-{    
-    my $auqRetVal;
+{
+    &usage() if (
+        !GetOptions(
+        'help|?'=>\&modeh,
+        'interactive'=>\&modei,
+        'list=s{1,2}'=>\&model,
+        'deploy=s{3,4}'=>\&moded));
+    if($mode eq 'i'){
+        &runmodei();
+    }
+    elsif($mode eq 'l'){
+        &runmodel();
+    }
+    elsif($mode eq 'd'){
+        &runmoded();
+    }
+    else{
+        &runmodeh();
+    }
+    
+ 
+};
+if($@)
+{
+    if ($@ !~ /EXIT_TOOL_NOW/)
+    {        
+        # Abnormal Termination ... 
+        print "$@";
+        &DeleteWorkingFolder();
+    }
+}
+# end of main subroutine
+
+sub usage
+{
+    ilog_print(1,"Unknown option:@_\n") if (@_);
+    ilog_print(1,"Usage: sudo perl $0 [-i|--interactive]\n");
+    ilog_print(1,"Usage: sudo perl $0 [-l|--list configfile]\n");
+    ilog_print(1,"Usage: sudo perl $0 [-d|--deploy  configfile siteindex publishprofile]\n");
+    ilog_print(1,"Usage: perl $0 [--help|-?]\n");
+    ilog_print(1,"configfile: The apache main config file\n");
+    ilog_print(1,"siteindex: The index of the source web site and should be number\n");
+    ilog_print(1,"publishprofile: The publishing profile file of the target azure web site\n");
+    exit 0;
+}
+
+sub modeh
+{
+    if($mode && $mode ne 'h'){&usage();};
+    $mode='h';
+    &utf_setRunMode($mode);
+}
+
+sub modei
+{
+   if($mode && $mode ne 'i') {&usage();};
+   $mode='i';
+   &utf_setRunMode($mode);
+}
+
+sub model
+{
+    if($mode && $mode  ne 'l') {&usage();}
```

Comment:
'l' is a bad name for a variable since it's hard to understand what it means and it looks like '1' or 'I'